.github/workflows/compatibility.yml

```diff
@@ -16,7 +16,7 @@ jobs:
       matrix:
         pack_kind: [current, previous]
         create_builder_kind: [current, previous]
-        lifecycle_kind: [current, previous]
+        lifecycle_kind: [default, previous]
```

Comment:
I believe (but happy to be shown incorrectly) that this change mainly represents the reality of what we're doing today.

The difference between `pack` & `lifecycle` was intentional. As this test suite is for `pack` it's important that we test the current code. It's not intended to test `main` of `lifecycle` because that repo offers fewer guarantees and could cause us to block changes in this repo unnecessarily.

The use of previous, and it's different meanings, also hasn't changed in this PR. The change here is to call out that we're using the `default` lifecycle for the latest `pack` code, rather than `current` which was also behaving differently between `pack` and `lifecycle`

Reply:
100%, my earlier comment was inaccurate. I just found it slightly confusing (when it came down to the `Fetcher`, where `previous` pack `relativeVersion` is still 0), but it makes sense.

Reply:
Ah, I missed this convo - some of this concern is duplicated in my other comments. The way `default` lifecycle works today is SO convoluted, thank goodness we are changing it. But as I understand it, when `default` is set we won't try to fill in the lifecycle path, resulting in a blank string being passed here: https://github.com/buildpacks/pack/blob/972ed1ef930df623bd8a0bdb9351519745b8f41f/acceptance/acceptance_test.go#L197. Therefore, ultimately, pack will download the lifecycle, and it will download the one that is specified in `builder.DefaultLifecycleVersion`. 

I think it would be nice if `default` meant `builder.DefaultLifecycleVersion`. If we could enable that through the inputs manager, even better.

Reply:
OK, so what I think we're describing here is that there could be a difference between `current` & `default`.

`current` should pick the latest GH release from the lifecycle repo. This accurately describes the existing implementation, as well as the like for like refactor that exists in this PR. Therefore `default` is objectively wrong in this case.

`default` would describe using `builder.DefaultLifecycleVersion` from the codebase being tested to determine which version to test. This is not currently being tested, although `builder.DefaultLifecycleVersion` _may_ correlate with the latest GH release for lifecycle repo, it may also correlate with `previous` or be a different version entirely.

As the purpose of this PR is refactoring, without changing testing functionality, I propose that I revert to using `current` & `previous` for lifecycle, removing all references to `default`. We should then explore separately what would be involved with adding the `default` concept for lifecycle to the suite.

How does that sound @natalieparellano @dfreilich @yaelharel ?

Reply:
Although, @natalieparellano's [suggestion](https://github.com/buildpacks/pack/pull/708#discussion_r445180541) of `latest` is really cool, and more accurate too.

Reply:
I agree that @natalieparellano's suggestion is better, but as @simonjjones mentioned, since the purpose of this PR is only refactoring, I think we should revert to using `current` and `previous` and create a new PR for making the change to `latest`.

Reply:
I'd be happy with keeping `current` vs. `latest` terminology for this PR, though I am glad you like my suggestion :)

I do think that `default` is currently being run in GHA [here](https://github.com/buildpacks/pack/blob/016be5a61a24b8ee0a9ef1f7a2c8292c95683828/.github/workflows/build.yml#L66) - because when you run acceptance without setting `LIFECYCLE_PATH` or `ACCEPTANCE_SUITE_CONFIG`, that's the code path we go down.